src/amo/components/CollectionManager/index.js

```diff
@@ -88,16 +99,30 @@ export class CollectionManagerBase extends React.Component<Props, State> {
       return;
     }
 
-    dispatch(updateCollection({
-      collectionSlug: collection.slug,
-      defaultLocale: collection.defaultLocale,
+    if (!this.state.slug) {
```

Comment:
I was going to gate this on `creating` too, but I don't think submitting a form without a `slug` is valid for either edit or create. If a user doesn't change the slug on the edit form, the current value will still be in `this.state.slug`, will it not?